Project/Assets/Scripts/Player/PlayerAttack.cs

```diff
@@ -69,30 +72,38 @@ public class PlayerAttack : MonoBehaviour {
             meleeAttackTrigger.enabled = false;
         }
     }
+    #endregion
 
-    private void CastAbility(Abilities ability)
+    //Casts players ability
+    private void CastAbility(AbilityTest ability)
     {
-        if (ability.isOffCooldown)
+        //Check to see ability is off cooldown
+        if (ability.offCooldown && (player.playerHUD.energySlider.value - ability.energy) > 0)
         {
-            anim.Play(ability.InputTag, 0);
+            //play the correct animation
+            anim.Play(ability.animationTag, 0);
+            //set the correct trigger
             meleeAttackTrigger.enabled = true;
-            meleeAttackTrigger.SendMessage("updateDamage", ability.Damage);
+            //update the triggers damage with abilities damage
+            meleeAttackTrigger.SendMessage("updateDamage", ability.damage);
+            player.playerHUD.PlayerCastedAbility(ability);
         }
         else
         {
             //print(ability.InputTag + " not off ability cooldown yet!");
         }
+        //update the players hud
+        
 
     }
 
-    IEnumerator CooldownHandler(Abilities ability)
+    //Coroutine used for ability cooldown
+    IEnumerator CooldownHandler(AbilityTest ability)
     {
         //print(ability.InputTag + " on " + ability.CoolDown+" second cooldown");
-        ability.isOffCooldown = false;
-        yield return new WaitForSeconds(ability.CoolDown);
-        ability.isOffCooldown = true;
+        ability.offCooldown = false;
+        yield return new WaitForSeconds(ability.cooldown);
```

Comment:
Can you double check for me this function can handle fraction of seconds? I'm looking at the documentation right now and it seems to, but just confirm for me :)